### PR TITLE
UCP/RKEY: Add check that rkey is used on same ep it's unpacked on

### PR DESCRIPTION
**Pull request by yosefe**
Returns error from shmem operations if invalid rkey is used in debug mode
follow up on  https://redmine.mellanox.com/issues/1582208 (Mellanox internal link)

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8372/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8374/ for details (Mellanox internal link).

File: src/ucp/core/ucp_mm.h [hunk 1]
```diff
@@ -173,6 +176,23 @@ ucp_memh2uct(ucp_mem_h memh, ucp_md_index_t md_idx)
         status; \
     })
 
+
+#if ENABLE_PARAMS_CHECK
+#define UCP_RKEY_RESOLVE(_rkey, _ep, _op_type) \
+    ({ \
+        ucs_status_t status = UCS_OK; \
```
Comment (brminich) [inline src/ucp/core/ucp_mm.h]: minor: looks like no need to init status here

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8376/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@hoopoepg @brminich pls take a look

**Pull request merged**